backend/redis/utils.go

```diff
@@ -492,7 +492,12 @@ func (r *Client) GetHubspotCompanies(ctx context.Context, companies interface{})
 }
 
 func (r *Client) SetGithubRateLimitExceeded(ctx context.Context, gitHubRepo string, expirationTime time.Time) error {
-	return r.setFlag(ctx, GithubRateLimitKey(gitHubRepo), true, time.Until(expirationTime))
+	expirationDuration := time.Until(expirationTime)
+	if expirationDuration <= 0 {
```

Comment:
are we sure that the `expirationTime` was in fact `0` when we were hitting that edge case, or could github have returned some duration that got formatted in a way that resulted in this being a huge number? could we add a unit test that mocks the response from github that was causing that behavior?

Reply:
Added some tests - I believe this is happening if an invalid time is provided to the Redis cache. This could be nil or a time in the past.

![Screenshot 2023-09-19 at 1 57 13 PM](https://github.com/highlight/highlight/assets/17744174/c761b95f-02bf-41f7-bd65-2be9c2c297c5)